rohmu/object_storage/google.py

```diff
@@ -259,26 +305,32 @@ def copy_file(self, *, source_key, destination_key, metadata=None, **_kwargs):
                 sourceBucket=self.bucket_name,
                 sourceObject=source_object,
             )
-            result = self._retry_on_reset(request, request.execute, operation=StorageOperation.copy_file)
+            reporter.report(self.stats)
```

Comment:
nit: I see quite some places where you call `Reporter()`, then `.report()` and then `_retry_on_reset()`. It could be combined together in one method

Reply:
Also, if we know all the possible operations, the reporter instances for these operations could be built-in (and cached?)
Like: `StorageOperation.copy_file.report(self.stats)`